external-crates/move/crates/move-package/src/resolution/resolution_graph.rs

```diff
@@ -325,9 +343,14 @@ impl Package {
     }
 
     fn define_addresses_in_package(&self, resolving_table: &mut ResolvingTable) -> Result<()> {
-        let package = self.source_package.package.name;
+        let pkg_name = custom_resolve_pkg_name(&self.source_package).with_context(|| {
```

Comment:
Is it worth encapsulating the `name` field and only exposing a function that will do the custom resolution?

Reply:
Will leave it up to you!  It might still be worth creating an abstraction for the name in its use as a key, and then add a separate abstraction to expose the user-friendly name.

Reply:
Let's leave this for last, I'll take a look at this when I do the name -> identifier renaming.